### PR TITLE
[ci] Set step level timeout

### PR DESCRIPTION
**Pull request by clee2000**
Not super important, but it is nice for the logs because the logs now say "the action timed out" instead of "the action was cancelled".  It also makes the job status "failure" instead of "cancelled"

also adds timeout minutes as an input for rocm and mac tests

### CODE REVIEW

**Comment by malfet** [comment]
I guess this supposed to be draft?

**Comment by clee2000** [comment]
> I guess this supposed to be draft?

is there anything more i should include? i was considering doing the JOB_TIMEOUT-30 step thing for mac and rocm too but they dont have timeout-minutes as an input so technically not required (although maybe it should)

**Comment by clee2000** [comment]
> > I guess this supposed to be draft?
> 
> is there anything more i should include? i was considering doing the JOB_TIMEOUT-30 step thing for mac and rocm too but they dont have timeout-minutes as an input so technically not required (although maybe it should)

nvm went and added timeout minutes as input for the linux mac and rocm

**Review by huydhn — APPROVED** [review]
LGTM!  It's nice to have timeout setup for macOS and ROCm.

File: .github/workflows/_rocm-test.yml [hunk 1]
```diff
@@ -120,7 +134,7 @@ jobs:
           XLA_CLANG_CACHE_S3_BUCKET_NAME: ossci-compiler-clang-cache-circleci-xla
           PYTORCH_TEST_CUDA_MEM_LEAK_CHECK: ${{ matrix.mem_leak_check && '1' || '0' }}
           PYTORCH_TEST_RERUN_DISABLED_TESTS: ${{ matrix.rerun_disabled_tests && '1' || '0' }}
-        timeout-minutes: 270
+        timeout-minutes: ${{ fromJson(steps.test-timeout.outputs.timeout) }}
```
Comment (huydhn) [inline .github/workflows/_rocm-test.yml:137]: I keep wondering what `fromJson` does here, converting string to number?

Comment (clee2000) [reply .github/workflows/_rocm-test.yml:137]: yep, if its not there lint complains about timeout-minutes needing a number but the step output is a string

File: .github/workflows/_linux-test.yml [hunk 2]
```diff
@@ -111,8 +111,17 @@ jobs:
         id: parse-ref
         run: .github/scripts/parse_ref.py
 
+      - name: Set Test step time
+        id: test-timeout
+        shell: bash
+        env:
+          JOB_TIMEOUT: ${{ inputs.timeout-minutes }}
+        run: |
+          echo "timeout=$((JOB_TIMEOUT-30))" >> "${GITHUB_OUTPUT}"
```
Comment (ZainRizvi) [inline .github/workflows/_linux-test.yml:120]: Is 30 a rough upper boundary for how many minutes may have elapsed in this job before this step runs?

**Pull request closed without merging**